projects/entity-data-api/src/Mirza/EntityDataAPI/Main.hs

```diff
@@ -0,0 +1,56 @@
+module Mirza.EntityDataAPI.Main (main) where
+
+import           Options.Applicative
+
+import           Network.HTTP.Client           (defaultManagerSettings,
+                                                newManager)
+
+import           Mirza.EntityDataAPI.AuthProxy (runAuthProxy)
+import           Mirza.EntityDataAPI.Types
+
+import           Network.HTTP.ReverseProxy     (ProxyDest (..))
+import qualified Network.Wai.Handler.Warp      as Warp
+
+import qualified Data.ByteString.Char8         as B
+
+import           Crypto.JOSE.JWK               (KeyMaterialGenParam (..),
+                                                genJWK)
+
+data Opts = Opts
+  { _myServiceInfo   :: ServiceInfo
+  , _destServiceInfo :: ServiceInfo
+  , _keySize         :: Int
+  }
+
+main :: IO ()
+main = launchProxy =<< execParser opts where
+  opts = info (optsParser <**> helper)
+    (fullDesc
+    <> progDesc "Reverse proxy for Mirza services"
+    <> header "Entity Data API")
+
+initContext :: Opts -> IO AuthContext
+initContext (Opts myService (destHost, destPort) kSize) = do
+  mngr <- newManager defaultManagerSettings
+  jwKey <- genJWK (RSAGenParam kSize)
+  let proxyDest = ProxyDest (B.pack destHost) destPort
+  pure $ AuthContext myService proxyDest mngr jwKey
+
+launchProxy :: Opts -> IO ()
+launchProxy opts = do
+  putStrLn "Initializing context..."
+  ctx <- initContext opts
+  putStrLn $ "Starting service on " <> (fst . myProxyServiceInfo $ ctx) <> ":" <> (show . snd . myProxyServiceInfo $ ctx)
+  Warp.run (fromIntegral . snd . myProxyServiceInfo $ ctx) (runAuthProxy ctx)
+
+optsParser :: Parser Opts
```

Comment:
It's typically easier to use environment variables instead of args when we're planning on deploying use Docker containers. Both would be nice if not too much work.